### PR TITLE
add parameter recover-cmd for process kill

### PR DESCRIPTION
**Pull request by FingerLeader**
Signed-off-by: FingerLeader <wanxfinger@gmail.com>
After setting `--recover-cmd`, the `recover-cmd` will be executed when a process kill experiment is recovered.

### CODE REVIEW

**Comment by FingerLeader** [comment]
All changes have been completed, PTAL @WangXiangUSTC

File: cmd/attack/process.go [hunk 1]
```diff
@@ -62,6 +62,7 @@ func NewProcessKillCommand(dep fx.Option, options *core.ProcessCommand) *cobra.C
 
 	cmd.Flags().StringVarP(&options.Process, "process", "p", "", "The process name or the process ID")
 	cmd.Flags().IntVarP(&options.Signal, "signal", "s", 9, "The signal number to send")
+	cmd.Flags().StringVarP(&options.RecoverCmd, "recover-cmd", "r", "", "The command to be executed at the end of experiment")
```
Comment (WangXiangUSTC) [inline cmd/attack/process.go:65]: how about changing the description to `The command to be executed when recovering experiment`?

File: pkg/server/chaosd/process.go [hunk 2]
```diff
@@ -65,13 +66,21 @@ func (processAttack) Recover(exp core.Experiment, _ Environment) error {
 	}
 	pcmd := config.(*core.ProcessCommand)
 	if pcmd.Signal != int(syscall.SIGSTOP) {
-		return core.ErrNonRecoverableAttack.New("only SIGSTOP process attack is supported to recover")
-	}
+		if pcmd.RecoverCmd == "" {
+			return core.ErrNonRecoverableAttack.New("only SIGSTOP process attack is supported to recover")
```
Comment (WangXiangUSTC) [inline pkg/server/chaosd/process.go:70]: need to update the error message, all process attack support recover

**Review by WangXiangUSTC — COMMENTED** [review]
rest LGTM

File: cmd/attack/process.go (hunk 1 above)
Comment (FingerLeader) [reply cmd/attack/process.go:65]: I agree with you, your description is more appropriate.

File: pkg/server/chaosd/process.go (hunk 2 above)
Comment (FingerLeader) [reply pkg/server/chaosd/process.go]: Error message has been updated

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

**Comment by WangXiangUSTC** [comment]
@Andrewmatilde PTAL

File: pkg/server/chaosd/process.go [hunk 3]
```diff
@@ -65,13 +66,21 @@ func (processAttack) Recover(exp core.Experiment, _ Environment) error {
...
 	if pcmd.Signal != int(syscall.SIGSTOP) {
-		return core.ErrNonRecoverableAttack.New("only SIGSTOP process attack is supported to recover")
-	}
+		if pcmd.RecoverCmd == "" {
+			return core.ErrNonRecoverableAttack.New("only SIGSTOP process attack and process attack with the recover-cmd are supported to recover")
+		}
 
-	for _, pid := range pcmd.PIDs {
-		if err := syscall.Kill(pid, syscall.SIGCONT); err != nil {
+		rcmd := exec.Command("bash", "-c", pcmd.RecoverCmd)
```
Comment (Andrewmatilde) [inline pkg/server/chaosd/process.go:73]: I suggest to log RecoverCmd here.

**Review by Andrewmatilde — COMMENTED** [review]
Rest, LGTM

File: pkg/server/chaosd/process.go (hunk 3 above)
Comment (FingerLeader) [reply pkg/server/chaosd/process.go:73]: I added a message output below.

**Review by Andrewmatilde — APPROVED** [review]
LGTM

**Pull request merged**